pkg/controllers/nodetemplate/controller.go

```diff
@@ -96,17 +107,41 @@ func (c *Controller) resolveSubnets(ctx context.Context, nodeTemplate *v1alpha1.
 	return nil
 }
 
-func (c *Controller) resolveSecurityGroup(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
+func (c *Controller) resolveSecurityGroups(ctx context.Context, nodeTemplate *v1alpha1.AWSNodeTemplate) error {
 	securityGroupIds, err := c.securityGroupProvider.List(ctx, nodeTemplate)
 	if err != nil {
 		return err
 	}
+	if len(securityGroupIds) == 0 {
+		logging.FromContext(ctx).Errorf("no security groups exist given constraints")
```

Comment:
Don't we already log for this as well? https://github.com/aws/karpenter/blob/main/pkg/providers/securitygroup/securitygroup.go#L108 I assume it works the same for subnets and AMIs, but I could be wrong.